apps/site/components/Common/Button/index.tsx

```diff
@@ -17,17 +24,49 @@ const Button: FC<ButtonProps> = ({
   href = undefined,
   children,
   className,
+  onClick,
   ...props
-}) => (
-  <Link
-    role="button"
-    href={disabled ? undefined : href}
-    aria-disabled={disabled}
-    className={classNames(styles.button, styles[kind], className)}
-    {...props}
-  >
-    {children}
-  </Link>
-);
+}) => {
+  //to handle the keyboard interactions, specifically for Enter and Space keys
+  const handleKeyDown = (e: KeyboardEvent<HTMLAnchorElement>) => {
+    if (disabled) {
+      e.preventDefault();
+      return;
+    }
+
+    if (e.key === 'Enter' || e.key === ' ') {
+      e.preventDefault();
+      if (typeof onClick === 'function') {
+        onClick(e as unknown as MouseEvent<HTMLAnchorElement>);
+      }
+    }
+  };
+
+  // to manage mouse click events for the component, providing behavior consistent with the disabled state
```

Comment:
I dont think these inline comments bring any value, it is quite evident already what these functions do :)